src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -63,15 +109,44 @@ var DOMRenderer = ReactFiberReconciler({
     newProps : Props,
     children : HostChildren<Instance>
   ) : boolean {
+    /*
+    Visualize the reconciliation
+    */
+    if (VISUALIZE_RECONCILIATION && typeof newProps.children === 'string') {
+      var c = +newProps.children;
+      if (!isNaN(c)) {
+        domElement.style.background = COLORS[c];
+      }
+    }
     return true;
   },
 
+  beginUpdate(domElement) {
+    if (VISUALIZE_RECONCILIATION) {
+      var c = (Math.round(Date.now() / 50) + 2) % COLORS.length;
+      if (!isNaN(c)) {
+        domElement.style.border = '3px solid ' + COLORS[c];
+      }
+    }
+  },
+
   commitUpdate(domElement : Instance, oldProps : Props, newProps : Props, children : HostChildren<Instance>) : void {
-    domElement.innerHTML = '';
-    recursivelyAppendChildren(domElement, children);
     if (typeof newProps.children === 'string') {
       domElement.textContent = newProps.children;
+      return;
+    }
+    if (typeof newProps.style === 'object') {
+      Object.assign(domElement.style, newProps.style);
```

Comment:
In the demo I intentionally avoid this Object.assign. This operation is slow enough that you'll drop frames. 731 updates x 0.02ms for this operation is enough.

It is absolutely critical that the commit path is super optimized and any slow work have to be done in the prepare step.